### PR TITLE
Features/add authz grant popup

### PR DESCRIPTION
**Pull request by Redm4x**
Added a Grant popup to make use of the new [Authorize Spend](https://docs.akash.network/testnet-technical-docs/authorized-spend) feature.

### CODE REVIEW

File: src/components/GrantModal/GrantModal.js [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+  };
+
+  const onSubmit = async ({ amount }) => {
+    setError("");
+    clearErrors();
+    const spendLimit = aktToUakt(amount);
+
+    const expirationDate = new Date(expiration);
+    const message = TransactionMessageData.getGrantMsg(address, granteeAddress, spendLimit, expirationDate);
+    const response = await sendTransaction([message]);
```
Comment (baktun14) [inline src/components/GrantModal/GrantModal.js:67]: try catch with success snackbar

File: src/components/GrantModal/GrantModal.js [hunk 2]
```diff
@@ -0,0 +1,193 @@
...
+              control={control}
+              name="expiration"
+              rules={{
+                required: true
+              }}
+              render={({ fieldState, field }) => {
+                return (
+                  <TextField
+                    {...field}
+                    type="datetime-local"
```
Comment (baktun14) [inline src/components/GrantModal/GrantModal.js:158]: Is the date UTC?

File: src/shared/utils/TransactionMessageData.js [hunk 3]
```diff
@@ -183,4 +184,31 @@ export class TransactionMessageData {
...
+            value: {
+              spend_limit: {
+                denom: "uakt",
+                amount: spendLimit.toString()
+              }
+            }
+          },
+          expiration: {
+            seconds: Math.floor(expiration.getTime() / 1_000),
+            nanos: Math.floor((expiration.getTime() % 1_000) * 1_000_000)
```
Comment (baktun14) [inline src/shared/utils/TransactionMessageData.js:206]: We can add some comments here to explain de math :)

File: src/components/GrantModal/GrantModal.js (hunk 2 above)
Comment (Redm4x) [reply src/components/GrantModal/GrantModal.js:158]: The input value is in local time, but when we convert it into a `Date` object and call `.getTime()` we get the UTC equivalent.

File: src/shared/utils/TransactionMessageData.js (hunk 3 above)
Comment (Redm4x) [reply src/shared/utils/TransactionMessageData.js]: Done

**Comment by baktun14** [comment]
image
Warnings

File: src/components/GrantModal/GrantModal.js (hunk 2 above)
Comment (baktun14) [reply src/components/GrantModal/GrantModal.js:158]: Ah ok good

**Review by baktun14 — APPROVED** [review]

**Pull request merged**